Move program.branch.Branch.branch_point to Repository

briannafan opened the pull request:
#### 77dd08dc75482c6a8b5afcc3e519df604f5cd2bb
<pre>
Move program.branch.Branch.branch_point to Repository
<a href="https://bugs.webkit.org/show_bug.cgi?id=262406">https://bugs.webkit.org/show_bug.cgi?id=262406</a>
rdar://problem/116256812

Reviewed by Jonathan Bedard.

Added branch_point() to git.py and refactored code to reflect change.

* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/git.py:
(Git):
(Git.branch_point):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py:
(Branch.editable):
(Branch.branch_point): Deleted.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull.py:
(Pull.main):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py:
(PullRequest.pull_request_branch_point):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/squash.py:
(Squash.main):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/branch_unittest.py:
* Tools/Scripts/webkitpy/common/checkout/scm/git.py:
(Git.remote_branch_ref):

Canonical link: <a href="https://commits.webkit.org/268734@main">https://commits.webkit.org/268734@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e939a513a29ec83abcf26b5072684af071cf9665

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/20523 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/26/builds/20947 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/21595 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/22419 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/19144 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/20756 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/23/builds/24203 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/16/builds/21124 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/22419 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/20742 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/23/builds/24203 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/17845 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/23276 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/20709 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/23/builds/24203 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/18650 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/23276 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/23/builds/24203 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/18826 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/23276 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/19414 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/16/builds/21124 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [❌ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/20344 "Failed resultsdbpy unit tests") | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/30/builds/18629 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/22968 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/2533 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/31/builds/19234 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

facetothefate reviewed (approved):
Need to remove MERGE_BASE_SHARD_SIZE in branch.py

Also need @JonWBedard to rubber stamp I think

gsnedders reviewed (commented):
This looks fine, in terms of just moving it.

I think we're going to need to make it take an argument at some point (probably a `rev`, rather than hardcoding `HEAD`), but just doing the move first seems fine to me.

Non-reviewer r+.

JonWBedard commented on Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/git.py:
```diff
@@ -620,6 +621,67 @@ def _is_on_default_branch(self, hash):
                 ], cwd=self.root_path, capture_output=True, encoding='utf-8').returncode == 0
         return default_branch in self.branches_for(hash)
 
+    def branch_point(self):
```
This should accept a `ref`, right? Whose default is `HEAD`?

briannafan replied:
Actually it never accepted a `ref`, but I needed to add that behavior for `git-webkit checkout` anyways.

JonWBedard reviewed (approved):
r+, with a minor change to `def branch_point` that should have been the original behavior.

briannafan committed:
Move program.branch.Branch.branch_point to Repository
https://bugs.webkit.org/show_bug.cgi?id=262406
rdar://problem/116256812

Reviewed by Jonathan Bedard.

Added branch_point() to git.py and refactored code to reflect change.

* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/git.py:
(Git):
(Git.branch_point):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py:
(Branch.editable):
(Branch.branch_point): Deleted.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull.py:
(Pull.main):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py:
(PullRequest.pull_request_branch_point):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/squash.py:
(Squash.main):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/branch_unittest.py:
* Tools/Scripts/webkitpy/common/checkout/scm/git.py:
(Git.remote_branch_ref):

Canonical link: https://commits.webkit.org/268734@main
